frontend/src/components/GradientRadialChart.tsx

```diff
@@ -0,0 +1,129 @@
+import { IconProp } from '@fortawesome/fontawesome-svg-core'
+import dynamic from 'next/dynamic'
+import { useTheme } from 'next-themes'
+import React from 'react'
+import { pluralize } from 'utils/pluralize'
+import SecondaryCard from 'components/SecondaryCard'
+
+// Importing Chart dynamically to avoid SSR issues with ApexCharts
+const Chart = dynamic(() => import('react-apexcharts'), {
+  ssr: false,
+})
+
+const GradientRadialChart: React.FC<{
+  title: string
+  icon?: IconProp
+  days: number
+  requirement: number
+}> = ({ title, days, icon, requirement }) => {
+  const { theme } = useTheme()
+  let checkpoint = requirement
+  let showRed = false
+  const stops = [0, 100]
+  const greenColor = '#0ef94e'
+  const redColor = '#f94e0e' // Red color for the end of the gradient
+  const orangeColor = '#f9b90e' // Orange color for the middle of the gradient
+  const colorStops = [
+    {
+      offset: 0,
+      color: greenColor,
+      opacity: 1,
+    },
+  ]
+  if (days > requirement) {
+    checkpoint = days
+    showRed = true
+  }
+  // Ensure checkpoint is at least 1 to avoid division by zero
+  checkpoint = checkpoint || 1
+  // Normalize days based on the checkpoint
+  const normalizedDays = (days / checkpoint) * 100
+  const normalizedRequirement = (requirement / checkpoint) * 100
+
+  if (showRed) {
+    const orangeStop = normalizedRequirement / 2
```

Comment:
I had this case when with 5 days requirement it still showed all green even though technically it should have showed at least orange. 
![Screenshot 2025-06-21 at 11 24 59 AM](https://github.com/user-attachments/assets/3180958a-649b-4b87-b92e-7b9c2852fe8d)
I can see we only set orange if `showRed` is truthy. Can we change this logic? Also, maybe we should start showing orange at 75% instead of 50%? I'm not sure, but would love to see how that looks with setups when project is almost at stale.